app/frontend/src/features/communities/events/EventPage.tsx

```diff
@@ -0,0 +1,211 @@
+      const attendanceStateToSet =
+        currentAttendanceState === AttendanceState.ATTENDANCE_STATE_GOING
+          ? AttendanceState.ATTENDANCE_STATE_NOT_GOING
+          : AttendanceState.ATTENDANCE_STATE_GOING;
+      return service.events.setEventAttendance({
+        attendanceState: attendanceStateToSet,
+        eventId,
+      });
+    },
+    {
+      onSuccess(updatedEvent) {
+        queryClient.setQueryData<Event.AsObject>(
+          eventKey(eventId),
+          updatedEvent
+        );
+        queryClient.invalidateQueries(eventKey(eventId));
+        queryClient.invalidateQueries({
+          predicate: ({ queryKey }) =>
+            queryKey[0] === eventAttendeesBaseKey &&
+            isEventUsersInputType(queryKey[1]) &&
+            queryKey[1].eventId === eventId,
+        });
+      },
+    }
+  );
+
+  useEffect(() => {
+    if (event?.slug && event.slug !== eventSlug) {
+      history.replace(routeToEvent(event.eventId, event.slug));
+    }
+  }, [event, eventSlug, history]);
+
+  return !isValidEventId ? (
+    <NotFoundPage />
+  ) : (
+    <>
+      {(eventError || setEventAttendanceError) && (
+        <Alert severity="error">
+          {eventError?.message || setEventAttendanceError?.message || ""}
+        </Alert>
+      )}
+      {isLoading ? (
+        <CircularProgress />
+      ) : (
+        event && (
+          <>
+            <div className={classes.header}>
+              <HeaderButton
+                className={classes.backButton}
+                onClick={() => history.goBack()}
+              >
+                <BackIcon />
+              </HeaderButton>
+              <div className={classes.eventTitle}>
+                <Typography variant="h1">{event.title}</Typography>
+                <Typography className={classes.eventTypeText} variant="body1">
+                  {event.onlineInformation
+                    ? VIRTUAL_EVENT
+                    : event.offlineInformation?.address}
+                </Typography>
+              </div>
+              <Button
+                className={classes.attendanceButton}
+                loading={isSetEventAttendanceLoading}
+                onClick={() => setEventAttendance(event.attendanceState)}
+              >
+                {event.attendanceState ===
+                AttendanceState.ATTENDANCE_STATE_GOING
+                  ? NOT_GOING
+                  : GOING}
+              </Button>
+              <div className={classes.eventTimeContainer}>
+                <CalendarIcon className={classes.calendarIcon} />
+                <div>
+                  <Typography variant="body1">
+                    {dayjs(timestamp2Date(event.startTime!)).format("LLLL")}
+                  </Typography>
+                  <Typography variant="body1">{TO}</Typography>
+                  <Typography variant="body1">
+                    {dayjs(timestamp2Date(event.endTime!)).format("LLLL")}
+                  </Typography>
+                </div>
```

Comment:
How does this work with timezones at the moment?
I think you should:
- have all the text inline
- make the width smaller so it's not just one line on wide screens
- make the "to" date only show the time if the date is the same

Reply:
I believe it uses the user computer's timezone at the moment as opposed to what's set on their profile, as I believe that will need to be set explicitly on the `dayjs` object if we want to prefer using what we got in our database instead. I will have to double check.

How do you think this should look on wider screens if it's not all going to be on one line? Wrap on the date like what I got now? That could be tricky to get right with everything all inline as the length could vary a lot depending on the localised format of how the date should look.

Reply:
Not sure about timezones. 

I think letting it wrap anywhere is fine, just make the width smaller